sklearn/linear_model/_base.py

```diff
@@ -575,6 +576,56 @@ def rmatvec(b):
         return self
 
 
+def _check_precomputed_gram_matrix(X, precompute, X_offset, X_scale,
+                                   rtol=1e-7,
+                                   atol=1e-5):
+    """Computes a single element of the gram matrix and compares it to
+    the corresponding element of the user supplied gram matrix.
+
+    If the values do not match a ValueError will be thrown.
+
+    Parameters
+    ----------
+    X : ndarray of shape (n_samples, n_features)
+        Data array.
+
+    precompute: array-like of shape (n_features, n_features)
```

Comment:
```suggestion
    precompute : array-like of shape (n_features, n_features)
```

Reply:
I assume that `precompute` should be an `ndarray` and not an `array-like` as specified in the docstring.

Reply:
in the ElasticNet constructor, the `precompute` param is also specified as `array-like`.

Reply:
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/_coordinate_descent.py#L611

Reply:
i've kept it as `array-like` for consistency, perhaps it should be updated in both places?